platform/Http.roc

```diff
@@ -0,0 +1,9 @@
+interface Http
+    exposes [
+        Request,
+        Response,
+    ]
+    imports []
+
+Request a : a | a has Decoding
+Response a : a | a has Encoding
```

Comment:
This sounds interesting. Is this the [roc ability feature](https://www.roc-lang.org/tutorial#abilities)?

I don't understand how this works. It says, it has to be a type, that can be decoded/encoded. But it does not say anything about the ecoding-format. For example, what happens, if I a `U128`. It may be not possible to encode it as json. But if I use another encoding-format, it might be possible.

Reply:
Also, I don't think it is possible to use a generic Request and Response here. The Request in the roc code has to be the same as the request in the go code.

On the other hand, you could use the generated wasm-file with different go-hosts (or other wasm-hosts). So a generic Request/Response could be useful

Reply:
With your proposal in mind, the wasm-platform could be even more generic. It does not have to be in the context of an http-server. You can send anything between a roc-wasm module and any wasm-runtime.

I merge you PR and then move some code around